Models/Terminal.php

```diff
@@ -74,11 +76,11 @@ public static function getActiveTerminals()
      *
      * @throws PrestaShopDatabaseException
      */
-    public static function getActiveTerminalsForCurrency($currency = false)
+    public static function getActiveTerminalsForCurrency($currency = false, $shop_id = 1)
     {
         return Db::getInstance(_PS_USE_SQL_SLAVE_)->ExecuteS('
             SELECT * FROM `'
-                                                             . _DB_PREFIX_ . "altapay_terminals` WHERE active = 1 AND currency = '" . $currency . "' ORDER BY IF(ISNULL(position), \"\", position) ASC, display_name DESC
+                                                             . _DB_PREFIX_ . "altapay_terminals` WHERE active = 1 AND currency = '" . $currency . "'AND shop_id = '" . (int) $shop_id . "' ORDER BY IF(ISNULL(position), \"\", position) ASC, display_name DESC
```

Comment:
If the shop_id has characters other than numbers, then casting it to `int` will return `0`, which in itself can be a real `shop_id`, returning results that are not desired.

Please perform proper validation and throw validation error in case the shop_id is not correct.

What about validating currency?

Reply:
I have updated this in the above comment and also validated the currency.